### DISCUSSION TITLE
datasets/detect/open-images-v7/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#7912] [Docs]
# datasets/detect/open-images-v7/

Dive into Google's Open Images V7, a comprehensive dataset offering a broad scope for computer vision research. Understand its usage with deep learning models.

https://docs.ultralytics.com/datasets/detect/open-images-v7/

**Comment by glenn-jocher** [comment]
Hey Ultralytics Users!

Exciting news! 🎉 We've added the Open Images V7 dataset to our collection. It's perfect for enhancing your YOLO models across various applications.

What's in it for you?

🌍 Diverse range of categories.
🎯 Accurate annotations for better training.
💡 Ideal for AI/ML advancements in computer vision.
Check out our new Open Images V7 page in the docs for all the details.

Questions or need help? Just ask us! We're here to support your YOLO journey.

Happy modeling! 🤖👀

**Comment by naga08krishna** [comment]
I only want to train 2 classes of images. Then how can I access OpenImagesV7.yaml for only two classes?

**Reply by glenn-jocher** [reply]
@naga08krishna to train a YOLO model on only two classes from the Open Images V7 dataset, you can modify the `open-images-v7.yaml` file to include only the classes you're interested in. Here’s how you can do it:

1. **Edit the YAML file**: Create a custom YAML file based on the original `open-images-v7.yaml` and include only the two classes you want to train on.

```yaml
# Custom Open Images V7 dataset for two classes
path: ../datasets/open-images-v7
train: images/train
val: images/val

# Classes
names:
  0: Airplane
  1: Banana
```

2. **Train the model**: Use this custom YAML file to train your model.

### Python Example

```python
from ultralytics import YOLO

# Load a COCO-pretrained YOLOv8n model
model = YOLO("yolov8n.pt")

# Train the model on the custom Open Images V7 dataset
results = model.train(data="path/to/custom-open-images-v7.yaml", epochs=100, imgsz=640)
```

### CLI Example

```bash
# Train a COCO-pretrained YOLOv8n model on the custom Open Images V7 dataset
yolo detect train data=path/to/custom-open-images-v7.yaml model=yolov8n.pt epochs=100 imgsz=640
```

This approach ensures that your training process focuses only on the specified classes. For more details, you can refer to the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

Feel free to reach out if you have any more questions! 😊

**Reply by naga08krishna** [reply]
Please tell me how to download the openimages.yaml file for only different types of vegetable images and how to download train and valid datasets? Tell me how to do this in colab.

**Reply by glenn-jocher** [reply]
Hi @naga08krishna,

To train a YOLO model on only vegetable images from the Open Images V7 dataset, you can create a custom YAML file that includes only the classes you're interested in. Here's how you can do it in Google Colab:

1. **Create a Custom YAML File**: Modify the `open-images-v7.yaml` to include only the vegetable classes you need. For example, if you want to include "Broccoli" and "Carrot", your YAML file would look like this:

```yaml
# Custom Open Images V7 dataset for vegetable classes
path: ../datasets/open-images-v7
train: images/train
val: images/val

# Classes
names:
  0: Broccoli
  1: Carrot
```

2. **Download and Prepare the Dataset**: Use the `fiftyone` library to download and filter the dataset for your specific classes. Here’s a Colab script to help you get started:

```python
# Install necessary libraries
!pip install fiftyone

import fiftyone as fo
import fiftyone.zoo as foz

# Define the classes you want to include
classes = ["Broccoli", "Carrot"]

# Load the Open Images V7 dataset
dataset = foz.load_zoo_dataset(
    "open-images-v7",
    split="train",
    label_types=["detections"],
    classes=classes,
    max_samples=1000,  # Adjust this number based on your needs
    dataset_dir="/content/open-images-v7"
)

# Export the dataset to YOLO format
dataset.export(
    export_dir="/content/open-images-v7-yolo",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="detections",
    classes=classes,
    split="train"
)
```

3. **Train the Model**: Use the custom YAML file to train your model.

### Python Example

```python
from ultralytics import YOLO

# Load a COCO-pretrained YOLOv8n model
model = YOLO("yolov8n.pt")

# Train the model on the custom Open Images V7 dataset
results = model.train(data="/content/open-images-v7-yolo/custom-open-images-v7.yaml", epochs=100, imgsz=640)
```

### CLI Example

```bash
# Train a COCO-pretrained YOLOv8n model on the custom Open Images V7 dataset
yolo detect train data=/content/open-images-v7-yolo/custom-open-images-v7.yaml model=yolov8n.pt epochs=100 imgsz=640
```

This approach ensures that your training process focuses only on the specified vegetable classes. For more details, you can refer to the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

Feel free to reach out if you have any more questions! 😊

**Comment by naga08krishna** [comment]
Thank you very much for answering my doubt.

**Reply by pderrenger** [reply]
@naga08krishna you're welcome! 😊

If you encounter any issues or bugs while training your model on the Open Images V7 dataset, please ensure you provide a minimum reproducible code example. This helps us understand and replicate the problem on our end. You can find more details on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, make sure you are using the latest versions of `torch` and `ultralytics`. If not, please update your packages and try again. This ensures you benefit from the latest fixes and improvements.

If you have any more questions or run into any issues, feel free to ask. We're here to help! 🚀

**Comment by naga08krishna** [comment]
Hi sorry to disturb you but please tell help me in this doubt
After downloading the images, when I try to convert the dataset into yolo
format I am getting below error repeatedly. Please help me in this.
[image: image.png]
[image: image.png]

And also tell me how to divide the dataset into train and validate datasets.

**Reply by pderrenger** [reply]
Hi @naga08krishna,

No worries, I'm here to help! 😊

### Error During Conversion
To assist you better with the error you're encountering while converting the dataset to YOLO format, could you please provide more details or a reproducible example? This will help us diagnose the issue more effectively. You can refer to our guide on creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example).

### Dataset Division
To divide your dataset into training and validation sets, you can use the `fiftyone` library, which simplifies the process. Here's a quick example to get you started:

```python
import fiftyone as fo
import fiftyone.zoo as foz

# Load the Open Images V7 dataset
dataset = foz.load_zoo_dataset(
    "open-images-v7",
    split="train",
    label_types=["detections"],
    dataset_dir="/path/to/open-images-v7",
    max_samples=1000  # Adjust this number based on your needs
)

# Split the dataset into train and validation sets
train_dataset = dataset.take(800)  # 80% for training
val_dataset = dataset.exclude(train_dataset)  # 20% for validation

# Export to YOLO format
train_dataset.export(
    export_dir="/path/to/open-images-v7-yolo/train",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="detections"
)
val_dataset.export(
    export_dir="/path/to/open-images-v7-yolo/val",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="detections"
)
```

### Training Example
Once your dataset is divided and converted, you can train your model using the following command:

```bash
yolo detect train data=open-images-v7.yaml model=yolov8n.pt epochs=100 imgsz=640
```

For more detailed information, you can visit our [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

Feel free to reach out if you have any more questions or need further assistance!

**Comment by naga08krishna** [comment]
I have converted the dataset into yolo format but all the images are in the
training set only. Please tell me how to divide the dataset into training
and validate datasets

**Reply by pderrenger** [reply]
Hi @naga08krishna,

Thank you for reaching out! To split your dataset into training and validation sets, you can use the `fiftyone` library, which provides a convenient way to handle and manipulate datasets. Here's a step-by-step guide to help you achieve this:

1. **Install FiftyOne**:
   Ensure you have the `fiftyone` library installed. You can install it using pip:
   ```bash
   pip install fiftyone
   ```

2. **Load and Split the Dataset**:
   Use the following script to load your dataset and split it into training and validation sets. This example assumes you have already converted the dataset into YOLO format and all images are currently in the training set.

   ```python
   import fiftyone as fo
   import fiftyone.zoo as foz
   from sklearn.model_selection import train_test_split

   # Load your dataset
   dataset_dir = "/path/to/your/yolo/dataset"
   dataset = fo.Dataset.from_dir(
       dataset_type=fo.types.YOLOv5Dataset,
       data_path=dataset_dir,
       labels_path=dataset_dir,
       include_id=True
   )

   # Split the dataset into training and validation sets
   train_view, val_view = dataset.take(0.8), dataset.take(0.2)

   # Export the training set
   train_view.export(
       export_dir="/path/to/save/train",
       dataset_type=fo.types.YOLOv5Dataset,
       label_field="ground_truth",
       classes=dataset.default_classes,
       split="train"
   )

   # Export the validation set
   val_view.export(
       export_dir="/path/to/save/val",
       dataset_type=fo.types.YOLOv5Dataset,
       label_field="ground_truth",
       classes=dataset.default_classes,
       split="val"
   )
   ```

3. **Update Your YAML File**:
   Ensure your `open-images-v7.yaml` file points to the correct directories for training and validation images:
   ```yaml
   path: ../datasets/open-images-v7
   train: /path/to/save/train/images
   val: /path/to/save/val/images
   ```

4. **Train Your Model**:
   Now you can train your model using the updated YAML file:
   ```python
   from ultralytics import YOLO

   # Load a COCO-pretrained YOLOv8n model
   model = YOLO("yolov8n.pt")

   # Train the model on the Open Images V7 dataset
   results = model.train(data="open-images-v7.yaml", epochs=100, imgsz=640)
   ```

   Or via CLI:
   ```bash
   yolo detect train data=open-images-v7.yaml model=yolov8n.pt epochs=100 imgsz=640
   ```

This should help you split your dataset and proceed with training. If you encounter any issues or need further assistance, feel free to ask! 😊

For more detailed information, you can refer to the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

**Comment by kaymanas** [comment]
I have downloaded and used the Ultralytics Open Images V7 Pretrained Models and they work great. However when I export to CoreML to use with XCode, the results are very different and much worse. There are many many more predictions and they're way off, the image attached predicted Snake, Treadmill, dumbell, Tent and many more at 99-100% confidence. 

See the images below for example of the output with the model as is and then converted to mlpackage format.  

https://imgur.com/a/bK2yc9a

Is there something wrong with the conversion process?

My python code here:

from ultralytics import YOLO

model = YOLO("yolov8x-oiv7.pt")  # load model

model.export(format="coreml",nms=True) # convert model

**Reply by pderrenger** [reply]
@kaymanas it sounds like the CoreML conversion might be causing issues with the model's performance. First, ensure you are using the latest version of the Ultralytics package. If the issue persists, try adjusting the `nms` parameter during export to see if it improves the results. Additionally, verify that the input image size and preprocessing steps in your XCode project match those used during training. If the problem continues, please provide more details or share minimal reproducible code snippets for further assistance.

**Reply by kaymanas** [reply]
Thanks @pderrenger. I've updated to the latest Ultralytics package (8.2.61).

I've tested exporting the model with nms=False. When reloaded into python and tested against a test image from the open image dataset it works perfectly. I cannot see the prediction when I load this into XCode unfortunately as there is no preview on the model page.

When I export with nms=True I cannot reload this into the python script as I get this error: "Ultralytics only supports inference of non-pipelined CoreML models exported with 'nms=False', but 'model=640' has an NMS pipeline created by an 'nms=True' export."

So I can only test this in XCode and it has a very poor prediction. Two image results here:
https://imgur.com/a/am7oF6Z

**Reply by pderrenger** [reply]
Thanks for updating to the latest version and testing the export options. The error message you're encountering indicates that the current implementation only supports non-pipelined CoreML models with `nms=False`. This limitation might be causing the poor predictions in XCode. To address this, ensure that the preprocessing steps in XCode match those used during training. If the issue persists, consider reporting it on our GitHub issues page for further investigation.

**Reply by kaymanas** [reply]
How do I ensure that the preprocessing steps in XCode match those used during training? Sorry I'm new to this, I tried googling but couldn't find any info...

**Reply by pderrenger** [reply]
@kaymanas to ensure preprocessing steps in XCode match those used during training, you need to replicate the same image resizing, normalization, and other preprocessing steps. Typically, this involves resizing images to the same dimensions used during training (e.g., 640x640), normalizing pixel values to the same range (e.g., 0-1), and applying any other transformations like mean subtraction or standard deviation scaling. You can find the exact preprocessing steps in the training script or model documentation. If you need further assistance, please refer to the [Ultralytics documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

**Reply by kaymanas** [reply]
Thanks for the help! where do I find the training script or model documentation? This page (the one you linked) doesn't have that information....

**Reply by pderrenger** [reply]
You can find the training script and model documentation in the Ultralytics YOLO repository on GitHub. Specifically, the [Ultralytics documentation](https://docs.ultralytics.com) provides comprehensive details on training, validation, and exporting models. For the Open Images V7 dataset, refer to the dataset's YAML configuration and the training section in the documentation. If you need further assistance, please check the repository's README or the relevant sections in the documentation.

**Comment by davidhuangal** [comment]
For the pretrained Oi7 models, what hyperparameters were used? Was it the `default.yaml` parameters? Or were things changed, like the number of epochs, etc.? I would like to train some YOLOv9 and YOLOv10 models on the dataset and wanted to make sure I was following the same training design as the YOLOv8 pretrained Oi7 models.

**Reply by glenn-jocher** [reply]
@davidhuangal the pretrained YOLOv8 models for the Open Images V7 dataset were trained using the `default.yaml` parameters. This includes standard hyperparameters and training settings. For your YOLOv9 and YOLOv10 models, you can start with these default settings and adjust as needed based on your specific requirements. If you encounter any issues or need further customization, please refer to the [Ultralytics documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/) for detailed guidance.

**Reply by glenn-jocher** [reply]
@davidhuangal ah one small change, epochs was lowered to 30 since OIv7 is so much larger than COCO. All else should be identical.

**Comment by nandiniigarg** [comment]
I am trying to train the YOLOv8m model using python for selected classes. Here's my YAML file:

```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# Open Images v7 dataset https://storage.googleapis.com/openimages/web/index.html by Google
# Documentation: https://docs.ultralytics.com/datasets/detect/open-images-v7/
# Example usage: yolo train data=open-images-v7.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── open-images-v7  ← downloads here (561 GB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: fiftyone/open-images-v7 # dataset root dir
train: train # train images (relative to 'path') 1743042 images
val: validation # val images (relative to 'path') 41620 images
test: # test images (optional)

# Classes
names:
  0: Bathroom accessory
  1: Bathroom cabinet
  2: Bathtub
  3: Bed
  4: Bench
.....
  159: Wine glass
  160: Wine rack
  161: Wood-burning stove

# Download script/URL (optional) ---------------------------------------------------------------------------------------
download: |
  from ultralytics.utils import LOGGER, SETTINGS, Path, is_ubuntu, get_ubuntu_version
  from ultralytics.utils.checks import check_requirements, check_version

  check_requirements('fiftyone')
  if is_ubuntu() and check_version(get_ubuntu_version(), '>=22.04'):
      # Ubuntu>=22.04 patch https://github.com/voxel51/fiftyone/issues/2961#issuecomment-1666519347
      check_requirements('fiftyone-db-ubuntu2204')

  import fiftyone as fo
  import fiftyone.zoo as foz
  import warnings
  classes_list = ["Bathroom accessory","Bathroom cabinet","Bathtub","Bed", "Bench", "Bidet", "Blender", "Book", "Bookcase", "Bowl", "Box", "Bronze sculpture>

  name = 'open-images-v7'
  fraction = 1.0  # fraction of full dataset to use
  LOGGER.warning('WARNING ⚠️ Open Images V7 dataset requires at least **561 GB of free space. Starting download...')
  for split in 'train', 'validation':  # 1743042 train, 41620 val images
      train = split == 'train'

      # Load Open Images dataset
      dataset = foz.load_zoo_dataset(name,
                                     split=split,
                                     label_types=['detections'],
                                     classes = classes_list,
                                     dataset_dir=Path(SETTINGS['datasets_dir']) / 'fiftyone' / name,
                                     max_samples=round((1743042 if train else 41620) * fraction))

      # Define classes
      if train:
          classes = dataset.default_classes  # all classes
          # classes = dataset.distinct('ground_truth.detections.label')  # only observed classes

      # Export to YOLO format
      with warnings.catch_warnings():
          warnings.filterwarnings("ignore", category=UserWarning, module="fiftyone.utils.yolo")
          dataset.export(export_dir=str(Path(SETTINGS['datasets_dir']) / name),
                         dataset_type=fo.types.YOLOv5Dataset,
                         label_field='ground_truth',
                         split='val' if split == 'validation' else split,
                         classes=classes,
                         overwrite=train)
```

I am running into this error after the first epoch:

```
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      7.34G          0      83.18          0          0        640:   7%|▋         | 1794/27205 [13:28<1:51:55,  3.78it/s]^[[B^[[B^[[B^[[B^[[B^[[B
      1/100      7.34G          0      5.485          0          0        640: 100%|██████████| 27205/27205 [1:52:47<00:00,  4.02it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 376/376 [02:20<00:00,  2.67it/s]
Traceback (most recent call last):
  File "/home/nirwana/nandini/open_images_test/ultralyticsTrial/train.py", line 7, in <module>
    results = model.train(data="open-images-v7.yaml", epochs=100, imgsz=640, device=0, patience=5, save=True, verbose=True, plots=True)
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/engine/model.py", line 813, in train
    self.trainer.train()
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/engine/trainer.py", line 206, in train
    self._do_train(world_size)
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/engine/trainer.py", line 430, in _do_train
    self.metrics, self.fitness = self.validate()
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/engine/trainer.py", line 575, in validate
    metrics = self.validator(self)
  File "/home/nirwana/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/engine/validator.py", line 193, in __call__
    stats = self.get_stats()
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/models/yolo/detect/val.py", line 181, in get_stats
    stats = {k: torch.cat(v, 0).cpu().numpy() for k, v in self.stats.items()}  # to numpy
  File "/home/nirwana/.local/lib/python3.10/site-packages/ultralytics/models/yolo/detect/val.py", line 181, in <dictcomp>
    stats = {k: torch.cat(v, 0).cpu().numpy() for k, v in self.stats.items()}  # to numpy
RuntimeError: torch.cat(): expected a non-empty list of Tensors
```

I cannot figure out how to fix this error.

**Reply by glenn-jocher** [reply]
@nandiniigarg it looks like the error might be due to an empty list of tensors during validation. Please ensure that your dataset and annotations are correctly formatted and that the selected classes have sufficient instances. Also, verify if the issue persists with the latest version of the Ultralytics package. For more details, you can refer to the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

**Reply by safwene-bahroun** [reply]
HELLO @glenn-jocher, i have the same error . Can you help me to fix it ?
![Capture d'écran 2025-03-09 110220](https://github.com/user-attachments/assets/7566cc14-5b20-4d06-8f94-6f758936bb98)
![Capture d'écran 2025-03-09 110206](https://github.com/user-attachments/assets/7aead159-007d-4729-b278-3a8d31f8d242)

**Reply by glenn-jocher** [reply]
@safwene-bahroun please provide a https://docs.ultralytics.com/help/minimum-reproducible-example/ to allow us to help you better.

**Reply by safwene-bahroun** [reply]
path: ../datasets/open-images-v7
train: C:\Users\GIGABYTE\datasets\open-images-v7\images\open-images-v7\train
val: C:\Users\GIGABYTE\datasets\open-images-v7\images\open-images-v7\validation
nc: 36
names:
  0: Person
  1: Man
  2: Woman
  3: Boy
  4: Belt
  5: Clothing
  6: Coat
  7: Cowboy hat
  8: Dress
  9: Fedora
  10: Footwear
  11: Glasses
  12: Glove
  13: Goggles
  14: Hat
  15: High heels
  16: Jeans
  17: Miniskirt
  18: Scarf
  19: Shirt
  20: Shorts
  21: Skirt
  22: Sock
  23: Sombrero
  24: Suit
  25: Sun hat
  26: Sunglasses
  27: Swim cap
  28: Swimwear
  29: Tie
  30: Trousers
  31: Cosmetics
  32: Face powder
  33: Hair spray
  34: Lipstick
  35: Perfume
download: |
  from ultralytics.utils import LOGGER, SETTINGS, Path, is_ubuntu, get_ubuntu_version
  from ultralytics.utils.checks import check_requirements, check_version
  import fiftyone as fo
  import fiftyone.zoo as foz
  from fiftyone import ViewField as F  # Import F for filtering
  import warnings

  # Define the list of classes
  classes = [
      "Person", "Man", "Woman", "Boy", "Belt", "Clothing", "Coat", "Cowboy hat", "Dress", "Fedora",
      "Footwear", "Glasses", "Glove", "Goggles", "Hat", "High heels", "Jeans", "Miniskirt", "Scarf",
      "Shirt", "Shorts", "Skirt", "Sock", "Sombrero", "Suit", "Sun hat", "Sunglasses", "Swim cap",
      "Swimwear", "Tie", "Trousers", "Cosmetics", "Face powder", "Hair spray", "Lipstick", "Perfume"
  ]

  # Dataset name and directory setup
  name = "open-images-v7"
  fo.config.dataset_zoo_dir = Path(SETTINGS["datasets_dir"]) / name / "images"

  # Log the start of the download process
  LOGGER.warning('Open Images V7 dataset: Starting download...')

  # Loop through train and validation splits
  for split in ['train', 'validation']:
      # Load Open Images dataset
      dataset = foz.load_zoo_dataset(
          name,
          split=split,
          classes=classes,
          label_types=['detections'],
          max_samples=5000 
      )

      # Define export directory
      export_dir = Path(SETTINGS['datasets_dir']) / name / "images" / ('train' if split == 'train' else 'validation')

      # Filter dataset to include only the specified classes
      filtered_dataset = dataset.match(F("ground_truth.detections.label").is_in(classes))

      # Export to YOLO format
      with warnings.catch_warnings():
          warnings.filterwarnings("ignore", category=UserWarning, module="fiftyone.utils.yolo")
          filtered_dataset.export(
              export_dir=str(export_dir),
              dataset_type=fo.types.YOLOv5Dataset,
              label_field='ground_truth',
              split=split,
              classes=classes,
              overwrite=True
          )

**Reply by glenn-jocher** [reply]
The `RuntimeError: torch.cat(): expected a non-empty list of Tensors` typically occurs when no validation predictions are generated. For Open Images V7 training with custom classes, please verify:

1. **Class Consistency**: Ensure your 36 selected classes exist in both train/val splits. Some fine-grained categories like "Man"/"Woman" might have limited val instances.
2. **Path Formatting**: Use forward slashes in paths: `train: datasets/open-images-v7/images/train`
3. **Annotations**: Confirm your filtered dataset contains labels by inspecting `labels/val` folder

For reference, see the Open Images V7 dataset setup guide in our documentation. Let us know if these checks resolve the validation issue.

**Comment by laila-e** [comment]
# Install necessary libraries
!pip install fiftyone

import fiftyone as fo
import fiftyone.zoo as foz

# Define the classes you want to include
classes = ["Broccoli", "Carrot"]

# Load the Open Images V7 dataset
dataset = foz.load_zoo_dataset(
    "open-images-v7",
    split="train",
    label_types=["detections"],
    classes=classes,
    max_samples=1000,  # Adjust this number based on your needs
    dataset_dir="/content/open-images-v7"
)

# Export the dataset to YOLO format
dataset.export(
    export_dir="/content/open-images-v7-yolo",
    dataset_type=fo.types.YOLOv5Dataset,
    label_field="detections",
    classes=classes,
    split="train"
)
I try this code and it works well but I need labels for detection and for visual relationship and I do not know how do that

**Reply by glenn-jocher** [reply]
@laila-e to include labels for both detection and visual relationships in your dataset, you'll need to adjust the `label_types` parameter in the `load_zoo_dataset` function to include both "detections" and "relationships". For more detailed guidance, please refer to the Open Images V7 documentation.

**Comment by ABINAYA200264** [comment]
Hello I'm Abinaya  I want selective classes in Open Images v7 dataset I use class Id to extract spacific classes  but it not work it detect all the the defult classes  I want only selected class, I don't have training spcilities  please can some one help me

**Comment by danielbob32** [comment]
Hi Glenn,

I noticed that while FiftyOne has support for YOLOv5Dataset, it doesn’t currently support instance segmentation, which I needed for training a custom YOLO model on a new class (lemons - don’t ask 😄).

After digging around, I couldn’t find a built-in API to export data in the format required by the segmentation models. So, I ended up creating a custom pipeline to handle the conversion. After a bit of fine-tuning, it worked quite well.

I’m wondering if this is something you’d be interested in adding here, and if so, what’s the best way to share it?

Also, I’d appreciate it if you could help me verify that my polygon annotations are formatted correctly.

Thanks!
Daniel

**Comment by DevanshL** [comment]
HI,can we finetune this OpenImages v7 dataset only for two classes car,person on yolov2 model.

**Reply by glenn-jocher** [reply]
Yes, but not in the current Ultralytics package with `YOLOv2`: if you only need `car` and `person`, you can filter Open Images V7 to those 2 classes and train a supported model like Ultralytics YOLO26 on that subset; for `YOLOv2` you’d need a separate legacy implementation. The [Open Images V7 docs](https://docs.ultralytics.com/datasets/detect/open-images-v7/) show the dataset layout and download setup, and feel free to keep follow-up questions here so the Ultralytics team can see them.
